### DISCUSSION TITLE
Confusion about Automatic Budgeting

### DISCUSSION THREAD

**Discussion by twiclo** [firefly-iii/firefly-iii#5035] [Firefly III] [answered]
A couple questions

1. The documentation says that there are two types of Automatic Budgeting; Fixed and Rollover. However when setting one up in the UI my choices are Fixed or "Add an amount every period" which I believe is the same as rollover but it's not clear
2. After reading the cron documentation it is also unclear to me which cron I need to call. Calling a command or triggering a recurring transaction? Does simply calling `/usr/bin/php /var/www/html/artisan firefly-iii:cron` run any scheduled task? Is it supposed to be set up to be called every time I want the budget to refresh or do I just call it every day so it picks up other scheduled tasks?

**Comment by JC5** [comment, marked as answer]
1. Yes, I'll make sure the documentation is fixed.
2. Yes, and it will only refresh the budget on the first day of the period, so run it at least daily.

**Reply by twiclo** [reply]
What goes in the "auto-budget amount" field when doing a rollover?

**Reply by JC5** [reply]
The amount? I'm not sure what you mean 😁

**Comment by viharm** [comment]
What happens if the cron fails to run on the first day of the month? Will it catch up and add the fixed monthly amount on next run? Or should I add the budgets for the "missed" month manually?

Want to check before I make a stupid mistake.

PS - Firefly III is awesome. Thanks for the hard work JC.

**Reply by JC5** [reply]
Yes, you will have to add them manually or run the cron job with `--force --date=2022-02-01` to create them.
